signing/signing.go

```diff
@@ -0,0 +1,105 @@
+package signing
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+
+	"github.com/ossf/scorecard-action/entrypoint"
+	"github.com/ossf/scorecard-action/options"
+	sigOpts "github.com/sigstore/cosign/cmd/cosign/cli/options"
+	"github.com/sigstore/cosign/cmd/cosign/cli/sign"
+)
+
+func SignScorecardResult(scorecardResultsFile string) error {
+	if err := os.Setenv("COSIGN_EXPERIMENTAL", "true"); err != nil {
+		return fmt.Errorf("error setting COSIGN_EXPERIMENTAL env var: %w", err)
+	}
+
+	// Prepare settings for SignBlobCmd.
+	rootOpts := &sigOpts.RootOptions{Timeout: sigOpts.DefaultTimeout} // Just the timeout.
+
+	keyOpts := sign.KeyOpts{
+		FulcioURL:    sigOpts.DefaultFulcioURL,     // Signing certificate provider.
+		RekorURL:     sigOpts.DefaultRekorURL,      // Transparency log.
+		OIDCIssuer:   sigOpts.DefaultOIDCIssuerURL, // OIDC provider to get ID token to auth for Fulcio.
+		OIDCClientID: "sigstore",
+	}
+	regOpts := sigOpts.RegistryOptions{} // Not necessary so we leave blank.
+
+	// This command will use the provided OIDCIssuer to authenticate into Fulcio, which will generate the
+	// signing certificate on the scorecard result. This attestation is then uploaded to the Rekor transparency log.
+	// The output bytes (signature) and certificate are discarded since verification can be done with just the payload.
+	if _, err := sign.SignBlobCmd(rootOpts, keyOpts, regOpts, scorecardResultsFile, true, "", ""); err != nil {
+		return fmt.Errorf("error signing payload: %w", err)
+	}
+
+	return nil
+}
+
+// Calls scorecard-api to process & upload signed scorecard results.
+// TODO: not sure how to test this because it requires running the entire scorecard action.
+func ProcessSignature() error {
+	// Get sarif output from file.
+	sarifPayload, err := ioutil.ReadFile(os.Getenv(options.EnvInputResultsFile))
+	if err != nil {
+		return fmt.Errorf("reading scorecard sarif results from file: %v", err)
+	}
+
+	// Change output settings to json and run scorecard again.
```

Comment:
add a note that this will be changed later to run scorecard only once, and generate multiple format. Maybe create a tracking issue for it.